staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -878,6 +889,15 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 	}
 }
 
+func (r *Request) injectTrace() opentracing.Span {
+	if r.ctx == nil {
+		r.ctx = context.Background()
+	}
+	span, ctx := opentracing.StartSpanFromContext(r.ctx, fmt.Sprintf("%v %v", r.verb, r.resource))
```

Comment:
It'd be way easier if there was a `nethttp.NewFromTracer(tr *opentracing.Tracer, opts...) Tracer` function in the library... Currently I don't think it's worth the effort.